### PR TITLE
Upstream documentation migration

### PR DESCRIPTION
**Pull request by wtrocki**
- Improvements to README
- Migrated wiki documentation from: https://github.com/feedhenry/fh-mbaas-api/wiki)

Issue #4

### CODE REVIEW

File: docs/Home.asciidoc [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+[[dataset-handler]]
+=== DataSet Handler
+
+A DataSet Handler is a function for intergrating the
+link:#sync-server[Sync Server] into a link:#dataset-backend[DataSet
+Backend]. +
+There are many handlers for doing CRUDL actions on a
+link:#dataset[DataSet] and managing collisions between
+link:#dataset-record[DataSet Records]. +
+The default implementation of these handlers uses fh.db (MongoDB backed
```
Comment (david-martin) [inline docs/Home.asciidoc]: This line makes me think this doc is either an out of date version (for old sync) or maybe not the right document to include

File: docs/Home.asciidoc [hunk 2]
```diff
@@ -0,0 +1,180 @@
+[[sync-framework-fh.sync]]
+= Sync Framework (fh.sync)
```
Comment (david-martin) [inline docs/Home.asciidoc]: I don't see the standalone version being referred to as fh.sync. I would see it just being sync, as you would probably do something like `var sync = require('fh-sync')` and not have any 'fh' object

File: docs/Sync Overview.asciidoc [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+= Data Sync Framework
+
+The {ProductShortName} mobile data synchronization framework provides the following key functionality:
+
+* Allows mobile apps to use & update data offline (local cache)
+* Provides a mechanism to manage bi-directional data synchronization from multiple client apps via app cloud & into back end data stores
+* Allows data updates (that is, deltas) to be distributed from the App Cloud to connected clients
+* Provides facilities for managing data collisions from multiple updates in the cloud
+
+This service effectively allows {ProductShortName} Apps to seamlessly continue working when the network connection is lost, and allow them to recover when the network connection is restored.
```
Comment (david-martin) [inline docs/Sync Overview.asciidoc]: This reads as too platform heavy for a standalone sync framework

File: docs/Home.asciidoc (hunk 1 above)
Comment (wtrocki) [reply docs/Home.asciidoc]: Good point. I missed that. Looks like wiki page is outdated to portal documentation:
https://github.com/feedhenry/fh-mbaas-api/wiki/ (was changed 3 months ago)

I will update it to recent version.

**Comment by wtrocki** [comment]
@david-martin - Reviewed documentation and removed lots of typos. Api documentation commit is optional - I think we should not commit it as it may be just generated on separate github pages branch :)

File: docs/Sync Server API.asciidoc [hunk 4]
```diff
@@ -1,18 +1,18 @@
 [[fh-sync]]
-= $fh.sync
+= sync
 
 The cloud sync framework requires handler functions to be defined which provide access to the back end data & manage data collisions. These are specified in the handleXXX() functions. Each unique dataset being managed is identified by a dataset_id which must be specified as the first parameter when calling any function of the sync framework.
 
-NOTE: Default implementations of the handler functions are already provided as part of the mBaas service. They use hosted db service to save data (via xref:fh-db[$fh.db]). If this is enough for your app, you don't need to implement the handler functions anymore, you just need to call the init function and provide sync options.
+NOTE: Default implementations of the handler functions are already provided as part of the mBaas service. They use hosted db service to save data (via xref:fh-db[db]). If this is enough for your app, you don't need to implement the handler functions anymore, you just need to call the init function and provide sync options.
```
Comment (wtrocki) [inline docs/Sync Server API.asciidoc]: Out of date. To be removed!

File: docs/Sync Client API.asciidoc [hunk 5]
```diff
@@ -742,7 +742,7 @@ var options = {
 
 // Parameters object to be passed to the cloud sync service.
 // It will be passed to the dataHandler when listing dataset on the back end.
-// If the default mBaas cloud implementation is used (which uses $fh.db for data handlers), all the valid list options can be used here.
+// If the default mBaas cloud implementation is used (which uses db for data handlers), all the valid list options can be used here.
```
Comment (wtrocki) [inline docs/Sync Client API.asciidoc]: This needs to be improved. fh.db => mongo db (using connectionURL)

**Comment by wtrocki** [comment]
@david-martin Can we merge that?

**Comment by david-martin** [comment]
@wtrocki I'd hold off on it until we figure out an upstream/downstream docs plan.
If we merged this, my concern is it would get out of date with the docs in the internal gitlab repo with CP docs

**Comment by wtrocki** [comment]
@david-martin We figured out plan to hold all on feedhenry.org. Until this will be available it's best to hold ascidocs directly in the repos so they can be maintained.

@secondsun Do you think we can merge this to this repo? We want to update documentation in some temporary place until actual documentaion on feedhenry.org will be available. We need to add some general improvements to documentation etc.

ping @damienomurchu @emilioicai

**Comment by damienomurchu** [comment]
@wtrocki +1 for a temporary home for all sync docs we will be looking to host on feedhenry.org eventually

**Comment by secondsun** [comment]
@wtrocki I think that hosting the docs for projects with the projects makes sense in the short term.  There are longer term things to figure out, but that is for the future.

**Comment by wtrocki** [comment]
Thank you so much lads! Going to improve documentation in separate PR's

**Pull request merged**